src/utils/validators.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  IResponseErrorValidation,
+  ResponseErrorValidation
+} from "@pagopa/ts-commons/lib/responses";
+import { pipe } from "fp-ts/lib/function";
+
+import * as express from "express";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { FiscalCode } from "../generated/FiscalCode";
+import { validIunList, validLegalFactList } from "../__mocks__/variables";
+import { LegalFactType } from "./types";
+import { IPNResponseErrorProblem, PNResponseErrorProblem } from "./responses";
+
+//this function should be replaced with the nearly next one, i'm not removing this at the moment to not break anything
+export const validateTaxIdHeader = (req: express.Request) => (): TE.TaskEither<
+  IResponseErrorValidation,
+  FiscalCode
+> =>
+  pipe(
+    req.headers["x-pagopa-cx-taxid"],
+    FiscalCode.decode,
+    E.mapLeft(() =>
+      ResponseErrorValidation(
+        "Error while processing request",
+        "Missing x-pagopa-cx-taxid header"
+      )
+    ),
+    TE.fromEither
+  );
+
+export const validateTaxIdInHeader = (
+  req: express.Request
+): TE.TaskEither<IPNResponseErrorProblem, FiscalCode> =>
+  pipe(
+    req.headers["x-pagopa-cx-taxid"],
+    FiscalCode.decode,
+    E.mapLeft(() =>
+      PNResponseErrorProblem(
+        "Invalid taxId",
+        "taxId must be provided inside headers",
+        []
+      )
+    ),
+    TE.fromEither
+  );
```

Comment:
They are pretty similar, maybe we can refactor them a little. 
As an exercise, you can even try to generalize the validation step with something like `validateTypeOrError`

Reply:
You are right those functions are the same but as mentioned in the comment right above, the `validateTaxIdHeader` should be removed because according to the specific when responding with a 400 we want to always use `PNResponseErrorProblem`. There are some other things to adjust and so I thought to make those changes in another pr